flink-streaming-connectors/flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java

```diff
@@ -0,0 +1,235 @@
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * A sink that delivers data to a Redis channel using the Jedis client.
+ * <p>When creating the sink using first constructor {@link #RedisSink(JedisPoolConfig, RedisMapper)}
+ * the sink will create connection using {@link redis.clients.jedis.JedisPool}.
+ * <p>When using second constructor {@link #RedisSink(JedisSentinelConfig, RedisMapper)} the sink will create connection
+ * using {@link redis.clients.jedis.JedisSentinelPool} to Redis cluster. Use this if redis is
+ * configured using sentinels else use the third constructor {@link #RedisSink(JedisClusterConfig, RedisMapper)}
+ * which use {@link redis.clients.jedis.JedisCluster} to connect to Redis cluster.
+ *
+ * <p>Example:
+ *
+ * <pre>
+ *{@code
+ *public static class RedisExampleDataMapper implements RedisMapper<Tuple2<String, String>> {
+ *    public RedisDataTypeDescription getDataTypeDescription() {
+ *        return new RedisDataTypeDescription(dataType, REDIS_ADDITIONAL_KEY);
+ *    }
+ *    public String getKeyFromData(Tuple2 data) {
+ *        return String.valueOf(data.f0);
+ *    }
+ *    public String getValueFromData(Tuple2 data) {
+ *        return String.valueOf(data.f1);
+ *    }
+ *}
+ *JedisPoolConfig jedisPoolConfig = new JedisPoolConfig.Builder()
+ *    .setHost(REDIS_HOST).setPort(REDIS_PORT).build();
+ *new RedisSink<String>(jedisPoolConfig, new RedisExampleDataMapper());
+ *}</pre>
+ *
+ * @param <IN> Type of the elements emitted by this sink
+ */
+public class RedisSink<IN> extends RichSinkFunction<IN> {
+
+	private static final long serialVersionUID = 1L;
+
+	private static final Logger LOG = LoggerFactory.getLogger(RedisSink.class);
+
+	/**
+	 * This additional key needed for {@link RedisDataType#HASH} and {@link RedisDataType#SORTED_SET}.
+	 * Other {@link RedisDataType} works only with two variable i.e. name of the list and value to be added.
+	 * But for {@link RedisDataType#HASH} and {@link RedisDataType#SORTED_SET} we need three variables.
+	 * <p>For {@link RedisDataType#HASH} we need hash name, hash key and element.
+	 * additionalKey used as hash name for {@link RedisDataType#HASH}
+	 * <p>For {@link RedisDataType#SORTED_SET} we need set name, the element and it's score.
+	 * additionalKey used as set name for {@link RedisDataType#SORTED_SET}
+	 */
+	private String additionalKey;
+	private RedisMapper<IN> redisSinkMapper;
+	private RedisDataType redisDataType;
+
+	private JedisPoolConfig jedisPoolConfig;
+	private JedisSentinelConfig jedisSentinelConfig;
+	private JedisClusterConfig jedisClusterConfig;
+
+	private RedisCommandsContainer redisCommandsContainer;
+
+	/**
+	 * Creates a new RedisSink that connects to the Redis Server.
+	 *
+	 * @param jedisPoolConfig The configuration of {@link org.apache.flink.streaming.connectors.redis.common.config.JedisPoolConfig}
+	 * @param redisSinkMapper This used for generate redis command and key value from incoming elements
```

Comment:
This is used to generate Redis command and key value from incoming elements.
Or
Used for generating Redis command and key value from incoming elements.